src/shared/sam/cli/samCliDeploy.ts

```diff
@@ -7,28 +7,41 @@
 
 import { getLogger, Logger } from '../../logger'
 import { ChildProcessResult } from '../../utilities/childProcess'
+import { map } from '../../utilities/collectionUtils'
 import { DefaultSamCliProcessInvoker } from './samCliInvoker'
 import { SamCliProcessInvoker } from './samCliInvokerUtils'
 
 export class SamCliDeployInvocation {
     public constructor(
         private readonly templateFile: string,
         private readonly stackName: string,
-        private readonly invoker: SamCliProcessInvoker =
-            new DefaultSamCliProcessInvoker(),
-        private readonly region: string
+        private readonly region: string,
+        private readonly parameterOverrides: Map<string, string>,
+        private readonly invoker: SamCliProcessInvoker = new DefaultSamCliProcessInvoker(),
     ) {
     }
 
     public async execute(): Promise<void> {
         const logger: Logger = getLogger()
-        const { exitCode, error, stderr, stdout }: ChildProcessResult = await this.invoker.invoke(
+
+        const args = [
             'deploy',
             '--template-file', this.templateFile,
             '--stack-name', this.stackName,
             '--capabilities', 'CAPABILITY_IAM',
-            '--region', this.region
-        )
+            '--region', this.region,
+        ]
+        if (this.parameterOverrides.size > 0) {
+            const overrides = [
+                ...map(
+                    this.parameterOverrides.entries(),
+                    ([key, value]) => `${key}=${value}`
+                )
+            ]
+            args.push('--parameter-overrides', ...overrides)
```

Comment:
Just making sure, the parameters don't need to be wrapped in anything?

Reply:
Correct, see the bottom of this page: https://docs.aws.amazon.com/cli/latest/reference/cloudformation/deploy/index.html